spec/duration.html

```diff
@@ -1398,67 +1398,39 @@ <h1>
       </dl>
       <emu-alg>
         1. Assert: If _plainRelativeTo_ is not *undefined*, _calendarRec_ is not *undefined*.
-        1. Let _allZero_ be *false*.
-        1. If _years_ = 0, and _months_ = 0, and _weeks_ = 0, and _days_ = 0, set _allZero_ to *true*.
-        1. If _largestUnit_ is *"year"* or _allZero_ is *true*, then
+        1. If _largestUnit_ is *"year"*, then
           1. Return ! CreateDateDurationRecord(_years_, _months_, _weeks_, _days_).
-        1. Let _sign_ be ! DurationSign(_years_, _months_, _weeks_, _days_, 0, 0, 0, 0, 0, 0).
-        1. Assert: _sign_ &ne; 0.
-        1. Let _oneYear_ be ! CreateTemporalDuration(_sign_, 0, 0, 0, 0, 0, 0, 0, 0, 0).
-        1. Let _oneMonth_ be ! CreateTemporalDuration(0, _sign_, 0, 0, 0, 0, 0, 0, 0, 0).
-        1. Let _oneWeek_ be ! CreateTemporalDuration(0, 0, _sign_, 0, 0, 0, 0, 0, 0, 0).
         1. If _largestUnit_ is *"month"*, then
           1. If _years_ = 0, return ! CreateDateDurationRecord(0, _months_, _weeks_, _days_).
           1. If _calendarRec_ is *undefined*, then
             1. Throw a *RangeError* exception.
           1. Assert: CalendarMethodsRecordHasLookedUp(_calendarRec_, ~dateAdd~) is *true*.
           1. Assert: CalendarMethodsRecordHasLookedUp(_calendarRec_, ~dateUntil~) is *true*.
-          1. Repeat, while _years_ &ne; 0,
-            1. Let _newRelativeTo_ be ? CalendarDateAdd(_calendarRec_, _plainRelativeTo_, _oneYear_).
-            1. Let _untilOptions_ be OrdinaryObjectCreate(*null*).
-            1. Perform ! CreateDataPropertyOrThrow(_untilOptions_, *"largestUnit"*, *"month"*).
-            1. Let _untilResult_ be ? CalendarDateUntil(_calendarRec_, _plainRelativeTo_, _newRelativeTo_, _untilOptions_).
-            1. Let _oneYearMonths_ be _untilResult_.[[Months]].
-            1. Set _plainRelativeTo_ to _newRelativeTo_.
-            1. Set _years_ to _years_ - _sign_.
-            1. Set _months_ to _months_ + _oneYearMonths_.
-          1. Return ? CreateDateDurationRecord(0, _months_, _weeks_, _days_).
+          1. Let _yearsDuration_ be ! CreateTemporalDuration(_years_, 0, 0, 0, 0, 0, 0, 0, 0, 0).
```

Comment:
Is the basic idea behind this change that we're delegating long-distance calculations to the calendar, where before we handled it via looping in the algorithm?  This seems OK to me, just want to make sure I understood the goal.

Reply:
Yes, that's correct!